### PR TITLE
Add ignore_self to memory creation

### PR DESCRIPTION
**Pull request by jacklanchantin**
# Description

When searching the memory, we sometimes want to ignore the self reference object. Previously, we had an option in the forward pass, which was deprecated. This change adds the ignore_self option as an argument to the memory searcher where the default option is False.

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [ ] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [x] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [ ] I want a deep design review.

## Before and After

If this PR introduces a change or fixes a bug, please add before and after screenshots (if this is causes a UX change) or before and after examples(this could be plain text, videos etc ) to demonstrate the change.

If this PR introduces a new feature, please also add examples or screenshots demonstrating the feature.

# Testing

I ran the build_memory_eqa_data.py script for the neural memory repo.

# Checklist:

- [x] I have performed manual end-to-end testing of the feature in my environment.
- [ ] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have added relevant collaborators to review the PR before merge.
- [ ] [Polymetis only] I have verified that all scripts in `tests/scripts` runs on hardware.

### CODE REVIEW

**Comment by jacklanchantin** [comment]
@aszlam I actually don't think this is necessary. We were previously using ignore_self, but I don't think it makes a difference right now. May be useful in the future though.

File: droidlet/memory/craftassist/mc_memory.py [hunk 1]
```diff
@@ -63,6 +63,7 @@ def __init__(
         agent_time=None,
         coordinate_transforms=None,
         agent_low_level_data={},
+        ignore_self=False,
```
Comment (aszlam) [inline droidlet/memory/craftassist/mc_memory.py:66]: why do we need this here? can't we just put it in the basic_search call ?

**Comment by aszlam** [comment]
If you do want to keep this, can you change it at https://github.com/facebookresearch/fairo/blob/caaa3e5541872de38e293e6495c0ce1f48869b92/droidlet/memory/memory_filters.py#L481 instead of in the agent memory?  like allow ignore_self to be passed in during a search, instead of at searcher init

**Comment by jacklanchantin** [comment]
I don't think we need this as of now, so I'm going to close this PR.

**Pull request closed without merging**